lib/components/popups/course-search/query-components/Form.tsx

```diff
@@ -15,68 +15,32 @@ import {
 } from '../../../../resources/commonTypes';
 import 'react-toastify/dist/ReactToastify.css';
 import Filters from './Filters';
-import {
-  selectCourseCache,
-  selectRetrievedAll,
-  updateCourseCache,
-} from '../../../../slices/userSlice';
 import axios from 'axios';
-import { getAPI } from '../../../../resources/assets';
-import { filterCourses } from './formUtils';
+import { getAPI, getParams } from '../../../../resources/assets';
 import { selectPlan } from '../../../../slices/currentPlanSlice';
 
-type SearchMapEl = {
-  course: SISRetrievedCourse;
-  version: number;
-  priority: number;
-};
-
 /**
  * Search form, including the search query input and filters.
  * TODO: filter by uppeer/lower levels
  *
  * @prop setSearching - sets searching state
  */
-const Form: FC<{ setSearching: (searching: boolean) => void }> = (props) => {
+const Form: FC<{
+  setSearching: (searching: boolean) => void;
+  pageIndex: number;
+  setPageCount: Function;
+  setPageIndex: Function;
+}> = ({ setSearching, pageIndex, setPageCount, setPageIndex }) => {
   // Set up redux dispatch and variables.
   const dispatch = useDispatch();
   const searchTerm = useSelector(selectSearchterm);
   const searchFilters = useSelector(selectSearchFilters);
   const semester = useSelector(selectSemester);
-  const courseCache = useSelector(selectCourseCache);
-  const retrievedAll = useSelector(selectRetrievedAll);
   const currentPlan = useSelector(selectPlan);
   const searchYear = useSelector(selectYear);
 
   // Component state setup
   const [showCriteria, setShowCriteria] = useState(false);
-  const [searchedCourses] = useState<Map<string, SearchMapEl>>(
-    new Map<string, SearchMapEl>(),
-  );
-  const [searchedCoursesFrequency] = useState<Map<string, number>>(
-    new Map<string, number>(),
-  );
-  const [initialQueryLength, setInitialQueryLength] = useState<number>(0);
-  const [searchedQuery, setSearchedQuery] = useState<any | null>(null);
-
-  // In order to make sure our search results correctly match our search term due to the asynchronous nature of finds.
-  // We use this hook to make sure that the search results are updated when the search term matches the term used for the search result.
-  useEffect(() => {
```

Comment:
no need because we do one api call and that's it. Search results load once and persists.